src/main/java/net/javadiscord/javabot/systems/qotw/submissions/SubmissionManager.java

```diff
@@ -91,68 +104,116 @@ public WebhookMessageCreateAction<?> handleSubmission(@NotNull ButtonInteraction
 		);
 		log.info("Opened new Submission Thread for User {}", member.getUser().getAsTag());
 		return Responses.success(event.getHook(), "Submission Thread created", "Successfully created a new private Thread for your submission.");
+	}
 
+	public List<QOTWSubmission> getActiveSubmissions() {
+		return config.getSubmissionChannel().getThreadChannels()
+				.stream()
+				.map(this::getOrRetrieveSubmission).toList();
 	}
 
 	/**
 	 * Handles the "Delete Submission" Button.
 	 *
 	 * @param event The {@link ButtonInteractionEvent} that is fired upon use.
 	 */
-	public void handleThreadDeletion(ButtonInteractionEvent event) {
-		ThreadChannel thread = (ThreadChannel) event.getGuildChannel();
-		asyncPool.execute(()->{
-			try {
-				Optional<QOTWSubmission> submissionOptional = qotwSubmissionRepository.getSubmissionByThreadId(thread.getIdLong());
-				if (submissionOptional.isPresent()) {
-					QOTWSubmission submission = submissionOptional.get();
-					if (submission.getAuthorId() != event.getMember().getIdLong()) {
-						return;
+	public void handleThreadDeletion(@NotNull ButtonInteractionEvent event) {
+		config.getSubmissionChannel().getThreadChannels()
+				.stream().filter(t -> t.getIdLong() == event.getChannel().getIdLong())
+				.map(this::getOrRetrieveSubmission)
+				.forEach(s -> getOrRetrieveAuthor(s, author -> {
+					if (event.getUser().getIdLong() == author.getIdLong()) {
+						s.getThread().delete().queue();
 					}
-					qotwSubmissionRepository.deleteSubmission(thread.getIdLong());
-					thread.delete().queue();
-				}
-			} catch (DataAccessException e) {
-				ExceptionLogger.capture(e, SubmissionManager.class.getSimpleName());
-			}
-		});
+				}));
+	}
+
+	private void getOrRetrieveAuthor(@NotNull QOTWSubmission submission, Consumer<User> onSuccess) {
+		if (submission.hasAuthor()) {
+			onSuccess.accept(submission.getAuthor());
+		} else {
+			submission.retrieveAuthor(author -> {
+				submission.setAuthor(author);
+				submissionCache.put(submission.getThread().getIdLong(), submission);
+				onSuccess.accept(author);
+			});
+		}
+	}
+
+	private QOTWSubmission getOrRetrieveSubmission(@NotNull ThreadChannel thread) {
+		if (submissionCache.containsKey(thread.getIdLong())) {
+			return submissionCache.get(thread.getIdLong());
+		} else {
+			QOTWSubmission submission = new QOTWSubmission(thread);
+			submissionCache.put(thread.getIdLong(), submission);
+			return submission;
+		}
 	}
 
 	private boolean canCreateSubmissions(Member member) {
 		if (member == null) return false;
 		if (member.getUser().isBot() || member.getUser().isSystem()) return false;
-		if (member.isTimedOut() || member.isPending()) return false;
-		return !hasActiveSubmissionThreads(member.getIdLong());
```

Comment:
Does that mean a member can create multiple submission threads?

Reply:
oh yeah, this needs to be fixed somehow

Reply:
fixed